### ISSUE TITLE
Network Tab not reporting on Nodes

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#19547]
We have a reported issue across a few versions of dotCMS where the network tab is not reporting on the current nodes that make up a cluster.  This has been seen in single servers, clusters and docker containers.  It is important that we get good reporting from the network tab in order to help us troubleshoot and resolve issues. 

The network tab looks like this:

![Screen Shot 2020-11-04 at 12 02 57 PM](https://user-images.githubusercontent.com/934364/98142593-fd167580-1e95-11eb-859e-6f94a2d4df65.png)

**Comment by wezell** [comment]
This happened because of the time change - we were using the current java date as the timestamp to know if we should delete an expired server from the `cluster_server` table in the `FreeServerFromClusterJob`.  When the time changed, the servers looked like they were over 1 hour without pinging and were deleted from the `cluster_server` table.

**Comment by nollymar** [comment]
I was not able to reproduce it. 
I tested the following scenarios in a cluster with one and two nodes: 
- dotCMS instances and database in the same host, moving forward/backward the host time
- dotCMS instances in the same host and external database, moving forward/backward the host time
- dotCMS instances in the same host and external database, moving forward/backward the database time
- dotCMS instances in different hosts and external database, moving forward/backward the host time

**Comment by bryanboza** [comment]
When you go to this tab, we need to list at least the local node, even when you don't have a cluster env configured, we need some work here... 

**Note:** 
Please test the heartbeat functionality here, in order to see if we are updating the information in real time.

**Comment by wezell** [comment]
@bryanboza I cannot reproduce what you are seeing.  I see the network tab and get the right data back.

**Comment by nollymar** [comment]
**Steps to reproduce:** 

1. Start a fresh installation with an enterprise license set.
2. Reset the license and switch between the Network Tab and License Tab several times before the license is automatically applied again.
3. When the license is assigned, move to the Network Tab and you should see the Nodes table empty.  

Try several times if you can't reproduce it because it is a race condition. The race condition makes the `cluster_server table` gets empty.

**Comment by wezell** [comment]
@nollymar this code in the ClusterFactory resolved it for me - the call to `APILocator.getServerAPI().getCurrentServer();` will call "createMyServer" if the server does not exist 

```java
    public static synchronized void rewireClusterIfNeeded() {
        if(!clusterReady()) {
            return;
        }
		try{
			List<Server> aliveServers = APILocator.getServerAPI().getAliveServers();
			Server server = APILocator.getServerAPI().getCurrentServer();
			
			boolean contains = aliveServers.contains(server);
			if(!aliveServers.equals(KNOWN_SERVERS) || !contains){

				Logger.info(ClusterFactory.class, "Cluster topography has changed. This might force a rewire");
				Logger.info(ClusterFactory.class, "Expecting Servers:" + KNOWN_SERVERS);
				Logger.info(ClusterFactory.class, "Found Servers:" + aliveServers);
				
				rewireCluster();
			}
		}
		catch(Exception e){
			Logger.error(ClusterFactory.class, "Unable to rewire cluster:" + e.getMessage());
			Logger.error(ClusterFactory.class, "servers:" + KNOWN_SERVERS);
			
			
			throw new DotRuntimeException(e);
		}
    }
```

**Comment by nollymar** [comment]
@wezell I tested your fix and worked for me as well :)

This is the PR: https://github.com/dotCMS/enterprise/pull/768

**Comment by nollymar** [comment]
Passed internal QA

**Comment by bryanboza** [comment]
Fixed, tested on master // Postgres // FF

**Comment by jcastro-dotcms** [comment]
The official code changes have been included in the dotCMS 5.3.8.4 LTS release only. The 5.2.8.4 LTS release is several commits behind, which made the code back-port not possible.
